include/dlaf/lapack/enum_output.h

```diff
@@ -37,7 +37,7 @@ inline std::ostream& operator<<(std::ostream& out, const Norm& norm) {
 }
 
 inline std::ostream& operator<<(std::ostream& out, const lapack::MatrixType& mtype) {
-  out << lapack::matrixtype2str(mtype);
+  out << to_c_string(mtype);
```

Comment:
In which version was this introduced? I.e. do we need a lower bound on blaspp/lapackpp versions?

Reply:
Good point!

This seems to be the [commit](https://github.com/icl-utk-edu/blaspp/commit/62ed424fbe0d675070160dbeeb6846dfe0fd4505) where they did the change, which is tagged with `2024.05.31`.

Looking at [2024.05.31 release notes](https://github.com/icl-utk-edu/blaspp/releases/tag/v2024.05.31), the change is actually reported

> Updated enum parameters to have to_string, from_string; deprecate <enum>2str, str2<enum>

I'm going to add the conflict to the spack package 😉

Reply:
@msimberg @rasolca 

do you think it is reasonable to use this?

```python
depends_on("blaspp@2024.05.31:", when="@0.6.1:")
```

We are going to probably release directly `0.7.0`, but in case we end up for some reason releasing a minor version, this would be more correct. Isn't it? We can also use `master`, but it looks more error-prone because we would have to remember to change that accordingly with the next release.

Any preference or suggestion?

Reply:
Isn't something like `depends_on("blaspp@:2024.05.31", when="@:0.6.0")` also needed?

Reply:
Yep, you're right! We can anticipate the actual deprecation too.

In the deprecation messages [here](https://github.com/icl-utk-edu/blaspp/blob/fb004632feddfbe4df6e06de8ec29406153b6e32/include/blas/util.hh#L175) and [here](https://github.com/icl-utk-edu/blaspp/blob/fb004632feddfbe4df6e06de8ec29406153b6e32/include/blas/util.hh#L275) they report `2025.05`, so probably we should add something like

```python
conflicts("^blasspp@2025.05:", when="@:0.6.0")
```

which differently from your suggestion would work even if they will make another 2024 release (and we should ship this PR with our next first release, either minor or major, but this sounds more than reasonable).

Reply:
> @msimberg @rasolca
> 
> do you think it is reasonable to use this?
> 
> ```python
> depends_on("blaspp@2024.05.31:", when="@0.6.1:")
> ```

Yep. We can always change it to `when="@0.7:"` when we decide what the next version will be.

> ```python
> conflicts("^blasspp@2025.05:", when="@:0.6.0")
> ```

:+1:

Reply:
see https://github.com/eth-cscs/DLA-Future/pull/1181/commits/7072da2199a1c351b572d79f4e3dadfd761b0480